flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/operations/converters/SqlShowDatabasesConverter.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.planner.operations.converters;
+
+import org.apache.flink.sql.parser.dql.SqlShowDatabases;
+import org.apache.flink.table.api.ValidationException;
+import org.apache.flink.table.catalog.CatalogManager;
+import org.apache.flink.table.operations.Operation;
+import org.apache.flink.table.operations.ShowDatabasesOperation;
+
+/** A converter for {@link SqlShowDatabases}. */
+public class SqlShowDatabasesConverter implements SqlNodeConverter<SqlShowDatabases> {
+
+    @Override
+    public Operation convertSqlNode(SqlShowDatabases sqlShowDatabases, ConvertContext context) {
+        if (sqlShowDatabases.getPreposition() == null) {
+            return new ShowDatabasesOperation(
+                    sqlShowDatabases.getLikeType(),
+                    sqlShowDatabases.getLikeSqlPattern(),
+                    sqlShowDatabases.isNotLike());
+        } else {
+            CatalogManager catalogManager = context.getCatalogManager();
+            String[] fullCatalogName = sqlShowDatabases.getCatalog();
+            if (fullCatalogName.length > 1) {
```

Comment:
Two things:

1.  I wonder if this check should be done sooner.
2. Does the client testing allow for dealing with errors?  If so, adding this case to the `catalog_database.q` would be nice.

Reply:
Yes, the `length` check is moved to the constructor of `SqlShowDatabases`, and I added a test case to catch the exception in `catalog_database.q `

Reply:
Cool.  @dawidwys does that make sense?  Or should we avoid having constructors throw exceptions?  (I'm not sure what the best practice is here.)